src/local-development/helpers/component-id-mapping-helper.ts

```diff
@@ -0,0 +1,77 @@
+import { getOriginObject } from './get-origin-object';
+import { LocalAppOrigin, MakecomappJson } from '../types/makecomapp.types';
+import { AppComponentType, AppGeneralType } from '../../types/app-component-type.types';
+
+/**
+ * Provides helping function to find remote component name from local ID and vice versa.
+ */
+export class ComponentIdMappingHelper {
+	constructor(private makecomappJson: MakecomappJson, private origin: LocalAppOrigin) {}
+
+	getRemoteName(componentType: AppComponentType | AppGeneralType, localId: string): string | null | undefined {
+		if (componentType === 'app') {
+			// This is special case, where "Common", "Readme" (etc.) are considered as members of virtual "app" compoment type, which has single component with ID ``.
+			return '';
+		}
+
+		const originInMakecomappJson = getOriginObject(this.makecomappJson, this.origin);
+
+		const matchedMappings =
+			originInMakecomappJson.idMapping?.[componentType].filter((mapping) => mapping.local === localId) ?? [];
+
+		if (matchedMappings.length === 0) {
+			return undefined;
+		}
+
+		if (matchedMappings.length >= 2) {
+			throw new Error(`Not unique mapping found for local ID "${localId}"`);
+		}
+
+		return matchedMappings[0].remote;
+	}
+
+	getRemoteNameStrict(componentType: AppComponentType | AppGeneralType, localId: string): string | null {
```

Comment:
can you comment why is this strict?

Reply:
Thanks for explanation. What about:

`getUnsyncedRemoteName()`
`getPendingRemoteName()`

standalone, solitary...

But I'm okay with strict too.

Reply:
Resolved as `getExistingRemoteName` vs `getRemoteName`